### PR TITLE
Partitioning should not be enabled when forwarding topology is used

### PR DESCRIPTION
**Pull request by SeanFeldman**
Fixes issue #131 

@Particular/azure-maintainers please review

Connects to #92

### CODE REVIEW

**Comment by yvesgoeleven** [comment]
change looks good, but needs rebase

**Comment by SeanFeldman** [comment]
@yvesgoeleven rebased and ready to go

File: src/Transport/Topology/PreStartupChecks/TopicPartitioningCheckForForwardingTopology.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿namespace NServiceBus.AzureServiceBus
+{
+    using System.Threading.Tasks;
+    using NServiceBus.Settings;
+    using NServiceBus.Transports;
+
+    class TopicPartitioningCheckForForwardingTopology
```
Comment (abombss) [inline src/Transport/Topology/PreStartupChecks/TopicPartitioningCheckForForwardingTopology.cs]: Why is this checked required, why would you disable partitioning on a forwarding topology?

Comment (SeanFeldman) [inline src/Transport/Topology/PreStartupChecks/TopicPartitioningCheckForForwardingTopology.cs]: ASB constraint.
https://github.com/Particular/NServiceBus.AzureServiceBus/issues/131#issuecomment-190350039

**Pull request merged**